### PR TITLE
Only count the users latest review

### PR DESCRIPTION
**Pull request by mweinelt**
Foxes Inflated review count due to
- approve
- comment/request changes
- approve

as seen in https://github.com/NixOS/nixpkgs/pull/176175

<s>Untested</s>

### CODE REVIEW

**Comment by mweinelt** [comment]
Oops, that won't work with later code.

File: src/label_approved/cli.py [hunk 1]
```diff
@@ -99,9 +99,19 @@ def main() -> None:
...
 
-        approved_reviews = [review for review in p_r_reviews if review.state == "APPROVED"]
-
-        approval_count = len(approved_reviews)
+        approvals = dict()
+        last_approved_review_date = datetime.min.date()
+        for review in p_r_reviews:
+            if review.state == "APPROVED":
+                approvals[review.user] = review
+                last_approved_review_date = review.submitted_at
```
Comment (mweinelt) [inline src/label_approved/cli.py:107]: This assumes the order of reviews is chronologically, which held true during my tests.

**Comment by mweinelt** [comment]
```pytb
>  label-approved
INFO:root:Pulls total: 8
INFO:root:Processing 176175
INFO:root:lastappdate: 2022-06-04 09:21:32
INFO:root:lastcommitdate: 2022-06-04 02:17:11
INFO:root:Removing label '12.approvals: 2' from PR: '176175' https://github.com/NixOS/nixpkgs/pull/176175
Traceback (most recent call last):
  File "/nix/store/h8s1yi7jvxcb6kblamiy1d8kzni6p3p6-python3.10-poetry2nix-env-scripts/bin/.label-approved-wrapped", line 13, in <module>
    sys.exit(main())
  File "/nix/store/v7kl295wwqxgax6gm3zw95aq5zwcgbzg-source/src/label_approved/cli.py", line 156, in main
    pr_object.p_r.remove_from_labels(label_to_remove)
  File "/nix/store/01h3ndmfxlgf6hklk422alxdzcljadak-python3-3.10.3-env/lib/python3.10/site-packages/github/PullRequest.py", line 809, in remove_from_labels
    headers, data = self._requester.requestJsonAndCheck(
  File "/nix/store/01h3ndmfxlgf6hklk422alxdzcljadak-python3-3.10.3-env/lib/python3.10/site-packages/github/Requester.py", line 353, in requestJsonAndCheck
    return self.__check(
  File "/nix/store/01h3ndmfxlgf6hklk422alxdzcljadak-python3-3.10.3-env/lib/python3.10/site-packages/github/Requester.py", line 378, in __check
    raise self.__createException(status, responseHeaders, output)
github.GithubException.GithubException: 403 {"message": "Must have admin rights to Repository.", "documentation_url": "https://docs.github.com/rest/reference/issues#remove-a-label-from-an-issue"}

**Comment by mweinelt** [comment]
```
> label-approved
INFO:root:Pulls total: 8
INFO:root:Processing 176175
INFO:root:lastappdate: 2022-06-04 09:21:32
INFO:root:lastcommitdate: 2022-06-04 02:17:11
INFO:root:Removing label '12.approvals: 2' from PR: '176175' https://github.com/NixOS/nixpkgs/pull/176175
INFO:root:Adding label '12.approvals: 1' to PR: '176175' https://github.com/NixOS/nixpkgs/pull/176175

**Review by Artturin — APPROVED** [review]
thanks, i forgot to turn on notifications for all events in this repo

**Pull request merged**